lib/puppet/provider/aem_crx_package/ruby.rb

```diff
@@ -109,10 +109,13 @@ def find_package
     path = "/etc/packages/#{@resource[:group]}/#{@resource[:pkg]}-.zip"
     begin
       retries ||= @resource[:retries]
+      retry_timeout = @resource[:retry_timeout]
       data = client.list(path: path, include_versions: true)
     rescue CrxPackageManager::ApiError => e
       Puppet.info("Unable to find package for Aem_crx_package[#{@resource[:pkg]}]: #{e}")
       will_retry = (retries -= 1) >= 0
+      Puppet.debug("Waiting #{retry_timeout} seconds before retrying package lookup") if will_retry
+      sleep retry_timeout
```

Comment:
If this is the last attempt, we won't want to sleep unless `will_retry` is true. So we might as well wrap the two debugging statements and this sleep in an `if` block.